internal/connor/eq.go

```diff
@@ -66,6 +69,27 @@ func eq(condition, data any) (bool, error) {
 	}
 }
 
+// objectsEqual returns true if the given condition and data
+// contain equal key value pairs.
+func objectsEqual(condition map[string]any, data any) (bool, error) {
+	if data == nil || condition == nil {
+		return reflect.DeepEqual(condition, data), nil
```

Comment:
todo: This looks very strange, and seems like an unusual place to justify a reflect call, can you please justify this, and if we are happy to keep as-is can you please document why you are doing this, as it doesn't seem to make any sense to me.

Reply:
I replaced it with a simpler nil comparison and I've added a test for this case